crypto/src/dex/lp/trading_function.rs

```diff
@@ -52,11 +55,42 @@ impl Protobuf<pb::TradingFunction> for TradingFunction {}
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(try_from = "pb::BareTradingFunction", into = "pb::BareTradingFunction")]
 pub struct BareTradingFunction {
+    /// The fee, expressed in basis points.
+    ///
+    /// The equation representing the fee percentage of the trading function (`gamma`) is:
+    /// `gamma = (10_000 - fee) / 10_000`.
     pub fee: u32,
     pub p: Amount,
     pub q: Amount,
 }
 
+impl BareTradingFunction {
+    /// Represent the trading function as a big-endian fixed point encoding
+    /// with 128 bits to the right of the decimal.
+    ///
+    /// Note: Currently this uses floating point to derive the encoding, which
+    /// is a placeholder and should be replaced by width-expanding polynomial arithmetic.
+    pub fn to_bytes(&self) -> [u8; 32] {
+        let effective_price = self.effective_price();
+        let integer = effective_price.trunc() as u128;
+        let fractional = effective_price.fract() as u128;
+
+        FixedEncoding::new(integer, fractional).to_bytes()
+    }
+
+    /// Returns the effective price of the trading function.
+    /// Note: the float math is a placehodler
+    pub fn effective_price(&self) -> f64 {
+        self.gamma() * self.p.inner as f64 / self.q.inner as f64
+    }
+
+    /// Returns the fee of the trading function, expressed as a percentage (`gamma`).
+    /// Note: the float math is a placehodler
+    pub fn gamma(&self) -> f64 {
+        (10_000.0 + self.fee as f64) / 10_000.0
```

Comment:
I think this should be `10000 - fee`, because `gamma` is multiplied into the amounts, so it should be in range `[0, 1]`, e.g., a 100bps = 1% fee should give `gamma = 0.99`

Reply:
I had it like that originally but it seemed weird that increasing the fee would decrease the effective price.

Reply:
It should definitely be `1 - f`, if increasing the fee decreases the effective price we've done a different mistake somewhere.  The percentage fee `gamma` is premultiplied by the input amounts `Delta_i`, to account for the LP position taking some fraction of the input as a fee: `Delta_i = (1 - f + f)*Delta = (1-f)*Delta + f*Delta = gamma*Delta + f*Delta`, where the left term is what's traded and the right term is what's taken as the fee.

Reply:
@zbuc I only skimmed the PR but note that the larger the `fee` the smaller the `gamma`:

e.g. `1% fee -> gamma = 0.99` or `30 bps fee -> gamma = 0.997` etc. 

So it's normal that with a larger gamma (i.e. smaller fee) the effective price would improve

Reply:
I guess I'm totally confused on what `effective_price` is expressing.

Reply:
It's the price that you can get in terms of asset `q` for asset `p` according to the trading function's parameters? So with a greater fee you get a worse price? That makes sense. I'll add a comment.